### PR TITLE
Fix moving torchmetrics to cuda device 0

### PR DESCRIPTION
**Pull request by awaelchli**
This PR removes `.to("cuda")` calls that cause the metric to be forcefully placed on GPU 0, causing a memory imbalance. A `torchmetrics` Metric is a `nn.Module`, and therefore it will be automatically placed on the right device when used in a LightningModule together with the Trainer.

The [best practice](https://lightning.ai/docs/pytorch/2.0.7/accelerators/accelerator_prepare.html#delete-cuda-or-to-calls) in Lightning is to not manipulate module or tensor placement unless there is a good reason for it.

### CODE REVIEW

**Review by exnx — APPROVED** [review]
looks great! Thank you :)
